File: shopy-api/Models/Review.cs
namespace shopy_api.Models;

public class Review : ISoftDeletable
{
    public Guid Id { get; set; }

    public Guid ProductId { get; set; }
    public Product Product { get; set; } = null!;

    public Guid UserId { get; set; }
    public ApplicationUser User { get; set; } = null!;

    // Diisi begitu ReviewsController & alur ulasan per-toko dibuat (Fase 7 TASKSELLER.md).
    public Guid? StoreId { get; set; }
    public Store? Store { get; set; }
    public Guid? SubOrderId { get; set; }
    public SubOrder? SubOrder { get; set; }

    public int Rating { get; set; }
    public string? Comment { get; set; }
    // JSON array string URL gambar ulasan, mis. ["url1","url2"] — cukup 1 kolom, belum perlu tabel terpisah.
    public string? ImageUrls { get; set; }

    public string? SellerReply { get; set; }
    public DateTime? SellerRepliedAt { get; set; }

    public DateTime CreatedAt { get; set; }
    public bool IsDeleted { get; set; }
}

File: shopy-mobile/lib/models/review/review.dart
/// Ulasan pembeli untuk 1 produk, sesuai `ReviewDto` di backend.
class Review {
  final String id;
  final String buyerName;
  final String? buyerAvatarUrl;
  final int rating;
  final String? comment;
  final List<String> imageUrls;
  final String? sellerReply;
  final DateTime? sellerRepliedAt;
  final DateTime createdAt;

  const Review({
    required this.id,
    required this.buyerName,
    this.buyerAvatarUrl,
    required this.rating,
    this.comment,
    this.imageUrls = const [],
    this.sellerReply,
    this.sellerRepliedAt,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] as String,
      buyerName: json['buyerName'] as String,
      buyerAvatarUrl: json['buyerAvatarUrl'] as String?,
      rating: json['rating'] as int,
      comment: json['comment'] as String?,
      imageUrls: (json['imageUrls'] as List? ?? []).map((e) => e as String).toList(),
      sellerReply: json['sellerReply'] as String?,
      sellerRepliedAt: json['sellerRepliedAt'] == null ? null : DateTime.parse(json['sellerRepliedAt'] as String),
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
